src/Processors/Transforms/buildPushingToViewsChain.cpp

```diff
@@ -223,6 +225,7 @@ class FinalizingViewsTransform final : public IProcessor
 /// Generates one chain part for every view in buildPushingToViewsChain
 std::optional<Chain> generateViewChain(
     ContextPtr context,
+    size_t view_level,
```

Comment:
Now the chain for a materialized view will be generated with view_level = 0 in any case, because `InterpreterInsertQuery` is used to generate the chain. 

Which means that for materialized views, this setting will be ignored. We probably need to include the level in the interpreter as well....